crates/web5_cli/src/doctor.rs

```diff
@@ -0,0 +1,218 @@
+use std::collections::HashMap;
+
+use crate::CheckType;
+
+pub struct HealthCheckState {
+    cli_version: Option<Option<String>>,
+    dependencies: Option<HashMap<String, bool>>,
+    env_vars: Option<HashMap<String, bool>>,
+    connectivity: Option<bool>,
+    basic_functionality: Option<bool>,
+}
+
+impl HealthCheckState {
+    fn new() -> Self {
+        Self {
+            cli_version: None,
+            dependencies: None,
+            env_vars: None,
+            connectivity: None,
+            basic_functionality: None,
+        }
+    }
+}
+
+fn check_cli_version() -> Option<Option<String>> {
+    Some(Some(env!("CARGO_PKG_VERSION").to_string())) // This will return the binary version.
+}
+
+fn check_dependencies() -> Option<HashMap<String, bool>> {
+    // TODO : Implement this function
+    // DOUBT  : Are we expecting to check system dependencies or dependencies in Cargo.toml ?
+    // If cargo dependencies then how exactly shall we check the versions ?
+    None
+}
```

Comment:
I think we dont need this, everything should be in the cli binary

Reply:
Got it. Will skip this then.